components/micro-gateway-cli/src/main/java/org/wso2/apimgt/gateway/cli/cmd/SetupCmd.java

```diff
@@ -146,209 +156,246 @@ public void execute() {
-        if (StringUtils.isEmpty(encryptedPass)) {
-            configuredTrustStorePass = null;
-        } else {
-            try {
-                configuredTrustStorePass = GatewayCmdUtils.decrypt(encryptedPass, password);
-            } catch (CliLauncherException e) {
-                //different password used to encrypt
-                configuredTrustStorePass = null;
+            //configure trust store
+            String configuredTrustStore = config.getToken().getTrustStoreLocation();
+            if (StringUtils.isEmpty(configuredTrustStore)) {
+                if (StringUtils.isEmpty(trustStoreLocation)) {
+                    isOverwriteRequired = true;
+                    if ((trustStoreLocation = promptForTextInput(
+                            "Enter Trust store location: [" + RESTServiceConstants.DEFAULT_TRUSTSTORE_PATH + "]")).trim()
+                            .isEmpty()) {
+                        trustStoreLocation = RESTServiceConstants.DEFAULT_TRUSTSTORE_PATH;
+                    }
+                }
+            } else {
+                trustStoreLocation = configuredTrustStore;
             }
-        }
 
-        if (StringUtils.isEmpty(configuredTrustStorePass)) {
-            if (StringUtils.isEmpty(trustStorePassword)) {
-                isOverwriteRequired = true;
-                if ((trustStorePassword = promptForPasswordInput("Enter Trust store password: [ use default? ]")).trim()
-                        .isEmpty()) {
-                    trustStorePassword = RESTServiceConstants.DEFAULT_TRUSTSTORE_PASS;
+            //configure trust store password
+            String encryptedPass = config.getToken().getTrustStorePassword();
+            String configuredTrustStorePass;
+            if (StringUtils.isEmpty(encryptedPass)) {
+                configuredTrustStorePass = null;
+            } else {
+                try {
+                    configuredTrustStorePass = GatewayCmdUtils.decrypt(encryptedPass, password);
+                } catch (CliLauncherException e) {
+                    //different password used to encrypt
+                    configuredTrustStorePass = null;
                 }
             }
-        } else {
-            trustStorePassword = configuredTrustStorePass;
-        }
 
-        File trustStoreFile = new File(trustStoreLocation);
-        if (!trustStoreFile.isAbsolute()) {
-            trustStoreLocation = GatewayCmdUtils.getUnixPath(GatewayCmdUtils.getCLIHome() + File.separator
-                    + trustStoreLocation);
-        }
-        trustStoreFile = new File(trustStoreLocation);
-        if (!trustStoreFile.exists()) {
-            logger.error("Provided trust store location {} does not exist.", trustStoreLocation);
-            throw new CLIRuntimeException("Provided trust store location does not exist.");
-        }
+            if (StringUtils.isEmpty(configuredTrustStorePass)) {
+                if (StringUtils.isEmpty(trustStorePassword)) {
```

Comment:
Do we need an else condition to handle the particular situation?

Reply:
This code segment is shown as added code because of the indentation. 
But, if the trust store password is not empty, we could proceed without needing an else condition right.